portfolio/src/main/java/com/google/sps/servlets/CommentService.java

```diff
@@ -0,0 +1,161 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.ZoneId;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+
+/** 
+ * This class contains the fields and methods
+ * required to use the Comment feature.
+ */
+ public class CommentService {
+
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+   
+  public List<Comment> getComments(PreparedQuery results, int limit) {
+    Iterable<Entity> newComments = results.asIterable(FetchOptions.Builder.withLimit(limit));    
+    
+    /** /data page is updated to contain JSON file of all images up to limit. */
+    List<Comment> comments = results.asList(FetchOptions.Builder.withLimit(limit))
+        .stream()
+        .map(entity -> {
+            
+            String postedTime = (String) entity.getProperty("postedTime");
+            String title = (String) entity.getProperty("title");
+            String text = (String) entity.getProperty("text");
+            String imageUrl = (String) entity.getProperty("imageUrl");
+
+            return new Comment(postedTime, title, text, imageUrl);
+        })
+        .collect(Collectors.toList());
+    return comments;
+
+  }
+  
+  /** Adds the comment to the datastore. */
+  public void saveComment(HttpServletRequest request) {
+    ZonedDateTime dateTime =  ZonedDateTime.now().withZoneSameInstant(ZoneId.of("America/New_York"));
```

Comment:
Generally it's recommended to store all time related fields in UTC if we can't keep them without a timezone. This way you can let the UI handle displaying the date appropriately. E.g. Showing the correct date to a user who is in Asia vs USA

Reply:
Okay, done.